journals/apps/core/tests/test_views.py

```diff
@@ -73,3 +76,33 @@ def test_setting_enabled(self):
 
         # Verify that the user has superuser permissions
         self.assertTrue(user.is_superuser)
+
+
+class TestUserRequireAuthView(TestCase):
+    """
+    Test Cases for required_auth
+    """
+
+    def setUp(self):
+        super(TestUserRequireAuthView, self).setUp()
+
+        self.user = UserFactory()
+        self.site = Site.objects.first()
+        self.site_configuration = SiteConfigurationFactory(site=self.site)
+        self.client.login(username=self.user.username, password='password')
+        self.path = reverse("require_auth")
+
+    def test_unauthorized_calls(self):
+        response = self.client.get(self.path, {"forward": "http://evil.com"})
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_authorized_calls(self):
```

Comment:
nit: can combine these two tests to one and make it ddt test?

Reply:
I am creating urls from dynamically from code for test. Could we do that using ddt ?
I am unable to find a way.
and also I think ddt makes separate test on run time for each data set. so does it not make less efficient ?

Reply:
@HammadAhmadWaqas Yeah you can get url like this
```
from django.test import RequestFactory
base_url = RequestFactory().get('/').build_absolute_uri()
```
Then prefix it to the urls coming from data of test like this
```
url = test_url.foramat(base_url=base_url)
```